simple_certmanager/admin.py

```diff
@@ -10,6 +10,7 @@
 
 @admin.register(Certificate)
 class CertificateAdmin(PrivateMediaMixin, admin.ModelAdmin):
+    model: type[Certificate]
```

Comment:
even though the model is defined as generic in both the mixin and the stubs for `ModelAdmin`, this is not picked up properly - I suspect because of the `@admin.register` usage which uses different type vars.

Declaring this type explicitly makes mypy accept it :shrug:

Reply:
Checked and the `register` decorator works with both mypy and pyright. It is not picked up properly because `PrivateMediaMixin` defines `model: type[Model]`, overriding the definition from `ModelAdmin` as it comes first in the mro.

And even then, to make it work properly you would have to inherit form `admin.ModelAdmin[Certificate]`, assuming you make it subscriptable (see https://github.com/lazybird/django-solo/pull/128#discussion_r1451790020). So it's a pain :/

Reply:
I've actually just patched django-privates to do the same thing as django-stubs with a type var: https://github.com/sergei-maertens/django-privates/blob/27c7391ef82479f5d99fb6542647ab999b3ebd2a/privates/admin.py#L46 so it no longer defines `model: type[Model]`

Also made the mixin generic and tried `PrivateMediaMixin[Certificate]`, which didn't yield any results. `admin.ModelAdmin` is not subscriptable out of the box, so the shit you have to do to make this pass is a bit excessive...